Sample.java

```diff
@@ -1,7 +1,71 @@
-// Time Complexity :
-// Space Complexity :
-// Did this code successfully run on Leetcode :
-// Any problem you faced while coding this :
-
+// Time Complexity : All operations except push - O(1), push operation - O(n)
+// Space Complexity : O(n)
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : I didn't think of maintaining a variable to keep track of the top element. 
+// I wrote it in the second version. Can you please mention in the comments later that how should we think of this from the beginning? 
```

Comment:
These little developments such as maintaining a front pointer would become more intuitive as we proceed along the course. Please make sure to whiteboard your problem really well, helps in arriving to a better solution.